### PR TITLE
feat: Add documentation examples for image preprocessing augmentation layers

### PR DESCRIPTION
**Pull request by Abhinavexists**
In reference to #21954
Added documentation examples across 20 image preprocessing augmentation layers.

### Updated Argumentation Layers
- [AugMix](https://keras.io/api/layers/preprocessing_layers/image_augmentation/aug_mix/)
- [CutMix](https://keras.io/api/layers/preprocessing_layers/image_augmentation/cut_mix/) 
- [MaxNumBoundingBox](https://keras.io/api/layers/preprocessing_layers/image_augmentation/max_num_bounding_boxes/)
- [RandAugment](https://keras.io/api/layers/preprocessing_layers/image_augmentation/rand_augment/)
- [RandomColorDegeneration](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_color_degeneration/)
- [RandomColorJitter](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_color_jitter/)
- [RandomContrast](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_contrast/)
- [RandomCrop](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_crop/)
- [RandomElasticTransform](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_elastic_transform/)
- [RandomErasing](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_erasing/)
- [RandomFlip](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_flip/)
- [RandomGaussianBlur](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_gaussian_blur/) 
- [RandomGrayscale](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_grayscale/)
- [RandomInvert](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_invert/)
- [RandomPerspective](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_perspective/)
- [RandomPosterization](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_posterization/)
- [RandomRotation](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_rotation/)
- [RandomSharpness](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_sharpness/)
- [RandomShear](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_shear/)
- [RandomTranslation](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_translation/)

### CODE REVIEW

File: keras/src/layers/preprocessing/image_preprocessing/rand_augment.py [hunk 1]
```diff
@@ -32,6 +32,26 @@ class RandAugment(BaseImagePreprocessingLayer):
...
             Options include `nearest`, `bilinear`. Default is `bilinear`.
         seed: Integer. Used to create a random seed.
+
+    Example:
+
+    ```python
+    (images, labels), _ = keras.datasets.cifar10.load_data()
+    images = images.astype("float32")
+
+    # Create a RandAugment layer
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/rand_augment.py]: To make the examples more concise, we can remove comments if the code is clear. For example, here the comment doesn't add much value as it's already clear what the code is doing.

File: keras/src/layers/preprocessing/image_preprocessing/rand_augment.py [hunk 2]
```diff
@@ -32,6 +32,26 @@ class RandAugment(BaseImagePreprocessingLayer):
...
+    augmenter = keras.layers.RandAugment(
+        value_range=(0, 255),
+        num_ops=2,  # Apply 2 random augmentation operations per image
+        factor=0.5  # Control the strength of augmentations (0-1 normalized)
+    )
+
+    # Apply augmentation during training
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/rand_augment.py]: same here regarding comment

File: keras/src/layers/preprocessing/image_preprocessing/max_num_bounding_box.py [hunk 3]
```diff
@@ -15,6 +15,29 @@ class MaxNumBoundingBoxes(BaseImagePreprocessingLayer):
...
+    bounding_boxes = {
+        "boxes": np.array([
+            [[10, 20, 100, 150], [50, 60, 200, 250], [0, 0, 50, 50]],
+        ]),
+        "labels": np.array([[1, 2, 3]])
+    }
+
+    # Ensure max 10 boxes per image
+    # If fewer than 10 boxes, pad with fill_value (-1)
+    # If more than 10 boxes, truncate to 10
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/max_num_bounding_box.py]: Could you make these comments a paragraph?

File: keras/src/layers/preprocessing/image_preprocessing/random_erasing.py [hunk 4]
```diff
@@ -41,6 +41,27 @@ class RandomErasing(BaseImagePreprocessingLayer):
...
         seed: Integer. Used to create a random seed.
+
+    Example:
+
+    ```python
+    # Create a RandomErasing layer with default factor
+    random_erasing = keras.layers.RandomErasing(factor=1.0)
+
+    # Your input image
+    image = [...]  # your input image
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_erasing.py]: Is it possible to create random images similar to previous examples?

File: keras/src/layers/preprocessing/image_preprocessing/random_invert.py [hunk 5]
```diff
@@ -30,6 +30,26 @@ class RandomInvert(BaseImagePreprocessingLayer):
...
+    random_invert = keras.layers.RandomInvert(factor=0.5)
+
+    # Your input image
+    image = [...]  # your input image
+
+    # Apply random color inversion
+    output = random_invert(image, training=True)
+
+    # For always inverting colors with custom value range
+    invert_always = keras.layers.RandomInvert(
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_invert.py]: Remove always from the variable name?

**Review by SamanehSaadat — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could you please review the comments in the examples and keep only the ones that are necessary? I think if the code is clear, it doesn't need a comment, e.g. "# Apply layer X" because it makes the example harder to read.

File: keras/src/layers/preprocessing/image_preprocessing/rand_augment.py [hunk 6]
```diff
@@ -32,6 +32,26 @@ class RandAugment(BaseImagePreprocessingLayer):
...
+    augmented_images = augmenter(images, training=True)
+
+    # At inference time, no augmentation is applied
+    output = augmenter(images, training=False)
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply keras/src/layers/preprocessing/image_preprocessing/rand_augment.py]: No need of import here

File: keras/src/layers/preprocessing/image_preprocessing/random_erasing.py (hunk 4 above)
Comment (Abhinavexists) [reply keras/src/layers/preprocessing/image_preprocessing/random_erasing.py]: i have added a batch of 4 image as a sample

**Comment by Abhinavexists** [comment]
Hi @SamanehSaadat i have made the changes as suggested by you, i hopes codes are much cleaner now while maintaining important comments.

File: keras/src/layers/preprocessing/image_preprocessing/cut_mix.py [hunk 7]
```diff
@@ -29,6 +29,30 @@ class CutMix(BaseImagePreprocessingLayer):
...
+    cutmix = keras.layers.CutMix(factor=0.8)
+
+    images = np.random.randint(0, 255, (8, 224, 224, 3), dtype='uint8')
+    labels = keras.ops.one_hot(
+        np.array([0, 1, 2, 3, 0, 1, 2, 3]),
+        num_classes=4
+    )
+
+    # Random rectangular patches are cut from one image and pasted into another
+    # Labels are also mixed proportionally to the patch area
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/cut_mix.py]: Please make this a paragraph.

File: keras/src/layers/preprocessing/image_preprocessing/aug_mix.py [hunk 8]
```diff
@@ -71,6 +71,25 @@ class AugMix(BaseImagePreprocessingLayer):
...
+        factor=0.3,
+        all_ops=True
+    )
+
+    images = np.random.randint(0, 255, (8, 224, 224, 3), dtype='uint8')
+
+    # Each image is augmented in 3 different ways (chains) and then mixed
+    augmented_images = augmix(images, training=True)
+
+    output = augmix(images, training=False)
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/aug_mix.py]: I think we can remove these second `output = ...` from all examples. One example of calling the layer should be enough.

File: keras/src/layers/preprocessing/image_preprocessing/random_contrast.py [hunk 9]
```diff
@@ -45,6 +45,21 @@ class RandomContrast(BaseImagePreprocessingLayer):
...
+
+    ```python
+    contrast_layer = keras.layers.RandomContrast(
+        factor=0.3,
+        value_range=(0, 255)
+    )
+
+    images = np.random.randint(0, 255, (2, 224, 224, 3), dtype='uint8')
+
+    contrasted_images = contrast_layer(images, training=True)
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_contrast.py]: We can probably remove extra white spaces between every line?

File: keras/src/layers/preprocessing/image_preprocessing/random_crop.py [hunk 10]
```diff
@@ -47,6 +47,20 @@ class RandomCrop(BaseImagePreprocessingLayer):
...
+
+    Example:
+
+    ```python
+    crop_layer = keras.layers.RandomCrop(height=192, width=192)
+
+    images = np.random.randint(0, 255, (2, 224, 224, 3), dtype='uint8')
+
+    cropped_images = crop_layer(images, training=True)
+    # Output shape: (2, 192, 192, 3)
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_crop.py]: Put the comment before the code?

File: keras/src/layers/preprocessing/image_preprocessing/random_gaussian_blur.py [hunk 11]
```diff
@@ -35,6 +35,28 @@ class RandomGaussianBlur(BaseImagePreprocessingLayer):
...
+    Example:
+
+    ```python
+    random_blur = keras.layers.RandomGaussianBlur(
+        factor=1.0,
+        kernel_size=3,
+        sigma=1.0
+    )
+
+    image = [...]  # your input image
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_gaussian_blur.py]: Can we make this one a random image too? like other examples

File: keras/src/layers/preprocessing/image_preprocessing/random_grayscale.py [hunk 12]
```diff
@@ -43,6 +43,20 @@ class RandomGrayscale(BaseImagePreprocessingLayer):
         Same as input shape. The output maintains the same number of channels
         as the input, even for grayscale-converted images where all channels
         will have the same value.
+
+    Example:
+
+    ```python
+    random_grayscale = keras.layers.RandomGrayscale(factor=0.5)
+
+    image = [...]  # your input image
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_grayscale.py]: same here

File: keras/src/layers/preprocessing/image_preprocessing/random_invert.py [hunk 13]
```diff
@@ -30,6 +30,22 @@ class RandomInvert(BaseImagePreprocessingLayer):
             represents the upper bound. Images passed to the layer should have
             values within `value_range`. Defaults to `(0, 255)`.
         seed: Integer. Used to create a random seed.
+
+    Example:
+
+    ```python
+    random_invert = keras.layers.RandomInvert(factor=0.5)
+
+    image = [...]  # your input image
```
Comment (SamanehSaadat) [inline keras/src/layers/preprocessing/image_preprocessing/random_invert.py]: same here

**Review by SamanehSaadat — CHANGES_REQUESTED** [review]
Thanks for making the changes. I think the PR looks better now. Left some minor feedback, mainly removing extra white spaces, removing calling layer with `Training=False` and using a random image instead of `[...]`.

**Review by hertschuh — COMMENTED** [review]
@Abhinavexists 

Thanks a lot for doing this!

Ideally, the example would cover all of these options:
```
{
  "images": images,
  "bounding_boxes": bounding_boxes,
  "labels": labels,
  "segmentation_masks": segmentation_masks,
} 
```
for all image preprocessing layers that do transform the image (not just change colors).

Sorry that's a lot of work. Maybe there's a way to prevent repetition by doing something like this: https://github.com/keras-team/keras/blob/master/keras/src/applications/efficientnet_v2.py#L1332-L1338

Or like this: https://github.com/keras-team/keras/blob/master/keras/src/optimizers/adam.py#L153

**Comment by Abhinavexists** [comment]
> @Abhinavexists 
> 
> Thanks a lot for doing this!
> 
> Ideally, the example would cover all of these options:
> ```
> {
>   "images": images,
>   "bounding_boxes": bounding_boxes,
>   "labels": labels,
>   "segmentation_masks": segmentation_masks,
> } 
> ```
> for all image preprocessing layers that do transform the image (not just change colors).
> 
> Sorry that's a lot of work. Maybe there's a way to prevent repetition by doing something like this: https://github.com/keras-team/keras/blob/master/keras/src/applications/efficientnet_v2.py#L1332-L1338
> 
> Or like this: https://github.com/keras-team/keras/blob/master/keras/src/optimizers/adam.py#L153

I will try my best to incorporate this way.

File: keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py [hunk 14]
```diff
@@ -383,3 +383,70 @@ def _compute_affine_matrix(
         )
 
         return affine_matrix
+
+
+base_image_preprocessing_transform_example = """
+```python
+layer = keras.layers.{LayerName}(bounding_box_format="xyxy")
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py]: no need as i said before

**Comment by Abhinavexists** [comment]
Hi @hertschuh and @SamanehSaadat 
Thanks a lot for pointing out for a more optimal approach towards these examples as it was becoming an redundant work across different layers 

as for the approach i went with https://github.com/keras-team/keras/blob/master/keras/src/optimizers/adam.py#L153 as i was able to understand the doc string format here completely and made much more sense in the context of this pr.

Now the `base_image_preprocessing_layer.py` has these `base_image_preprocessing_transform_example`, `base_image_preprocessing_example`, `base_image_preprocessing_color_example` which are used as shared templates


Only one case is that in `max_num_bounding_box.py` i had to keep an explicit example as it only using bounding box.

File: keras/src/layers/preprocessing/image_preprocessing/random_grayscale.py [hunk 15]
```diff
@@ -3,6 +3,9 @@
 from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
     BaseImagePreprocessingLayer,
 )
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    base_image_preprocessing_example,
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/random_grayscale.py]: This one should use `base_image_preprocessing_color_example`.

File: keras/src/layers/preprocessing/image_preprocessing/random_invert.py [hunk 16]
```diff
@@ -2,6 +2,9 @@
 from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
     BaseImagePreprocessingLayer,
 )
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    base_image_preprocessing_example,
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/random_invert.py]: This one should use `base_image_preprocessing_color_example`.

File: keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py [hunk 17]
```diff
@@ -383,3 +383,70 @@ def _compute_affine_matrix(
...
+        "bounding_boxes": bounding_boxes,
+        "labels": labels,
+        "segmentation_masks": segmentation_masks
+    },
+    training=True
+)
+```
+"""
+
+base_image_preprocessing_example = """
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py]: I don't think we need `base_image_preprocessing_example`, we can always use the other two. I commented in the other files about this.

File: keras/src/layers/preprocessing/image_preprocessing/aug_mix.py [hunk 18]
```diff
@@ -5,6 +5,9 @@
 from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
     BaseImagePreprocessingLayer,
 )
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    base_image_preprocessing_example,
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/aug_mix.py]: This layer does transforms, so we should use `base_image_preprocessing_transform_example`.

File: keras/src/layers/preprocessing/image_preprocessing/rand_augment.py [hunk 19]
```diff
@@ -3,6 +3,9 @@
 from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
     BaseImagePreprocessingLayer,
 )
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    base_image_preprocessing_example,
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/rand_augment.py]: This layer does transforms, so we should use `base_image_preprocessing_transform_example`.

**Comment by Abhinavexists** [comment]
@hertschuh removed `base_image_preprocessing_example`

**Review by hertschuh — APPROVED** [review]
